src/rotation.rs

```diff
@@ -0,0 +1,645 @@
+// Copyright 2013 The Servo Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use approxeq::ApproxEq;
+use num_traits::{Float, One, Zero};
+use std::fmt;
+use std::ops::{Add, Neg, Mul, Sub, Div};
+use std::marker::PhantomData;
+use trig::Trig;
+use {TypedPoint2D, TypedPoint3D, TypedVector2D, TypedVector3D, Vector3D, point2, point3, vec3};
+use {TypedTransform3D, TypedTransform2D, UnknownUnit, Radians};
+
+
+define_matrix! {
+    /// A transform that can represent rotations in 2d, represented as an angle in radians.
+    pub struct TypedRotation2D<T, Src, Dst> {
+        pub angle : T,
+    }
+}
+
+/// The default 2d rotation type with no units.
+pub type Rotation2D<T> = TypedRotation2D<T, UnknownUnit, UnknownUnit>;
+
+impl<T, Src, Dst> TypedRotation2D<T, Src, Dst> {
+    #[inline]
+    /// Creates a rotation from an angle in radians.
+    pub fn new(angle: Radians<T>) -> Self {
+        TypedRotation2D {
+            angle: angle.0,
+            _unit: PhantomData,
+        }
+    }
+
+    /// Creates the identity rotation.
+    #[inline]
+    pub fn identity() -> Self where T: Zero {
```

Comment:
this is begging for an associated constant

Reply:
True that. Let's add it for this type and the other ones as a followup.